commit 7528344cc93d71c65dd26c742a8e641f2927940e
Author: Remy Maucherat <remm@apache.org>
Date:   Wed Feb 18 09:25:30 2015 +0000

    - Add debugging of my own (sorry), on socket close.
    - Try dropping the direct socket close from the upgraded streams (the sockets seem to still be closed and the ws tests pass with NIO1+2). Actually following the discussions on the EG, I don't think the close required logic is fully right, it would seem it should wait for both input AND output to be closed, but this needs to be validated.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1660582 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/coyote/http11/upgrade/UpgradeServletInputStream.java b/java/org/apache/coyote/http11/upgrade/UpgradeServletInputStream.java
index 5ba3bc6..69152f2 100644
--- a/java/org/apache/coyote/http11/upgrade/UpgradeServletInputStream.java
+++ b/java/org/apache/coyote/http11/upgrade/UpgradeServletInputStream.java
@@ -142,7 +142,6 @@ public class UpgradeServletInputStream extends ServletInputStream {
     @Override
     public void close() throws IOException {
         closeRequired = true;
-        socketWrapper.close();
     }
 
 
diff --git a/java/org/apache/coyote/http11/upgrade/UpgradeServletOutputStream.java b/java/org/apache/coyote/http11/upgrade/UpgradeServletOutputStream.java
index 76c13c9..5264990 100644
--- a/java/org/apache/coyote/http11/upgrade/UpgradeServletOutputStream.java
+++ b/java/org/apache/coyote/http11/upgrade/UpgradeServletOutputStream.java
@@ -172,7 +172,6 @@ public class UpgradeServletOutputStream extends ServletOutputStream {
     @Override
     public void close() throws IOException {
         closeRequired = true;
-        socketWrapper.close();
     }
 
 
diff --git a/java/org/apache/tomcat/util/net/Nio2Endpoint.java b/java/org/apache/tomcat/util/net/Nio2Endpoint.java
index 6cc1248..b7a5d3d 100644
--- a/java/org/apache/tomcat/util/net/Nio2Endpoint.java
+++ b/java/org/apache/tomcat/util/net/Nio2Endpoint.java
@@ -589,6 +589,10 @@ public class Nio2Endpoint extends AbstractEndpoint<Nio2Channel> {
     }
 
     public void closeSocket(SocketWrapperBase<Nio2Channel> socket) {
+        if (log.isDebugEnabled()) {
+            log.debug("Calling [" + this + "].closeSocket([" + socket + "],[" + socket.getSocket() + "])",
+                    new Exception());
+        }
         if (socket == null) {
             return;
         }
